airflow/providers/google/cloud/sensors/workflows.py

```diff
@@ -0,0 +1,123 @@
+    :type location: str
+    :param success_states: Execution states to be considered as successful, by default
+        it's only ``SUCCEEDED`` state
+    :type success_states: List[Execution.State]
+    :param failure_states: Execution states to be considered as failures, by default
+        they are ``FAILED`` and ``CANCELLED`` states.
+    :type failure_states: List[Execution.State]
+    :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+        retried.
+    :type retry: google.api_core.retry.Retry
+    :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+        ``retry`` is specified, the timeout applies to each individual attempt.
+    :type timeout: float
+    :param metadata: Additional metadata that is provided to the method.
+    :type metadata: Sequence[Tuple[str, str]]
+    """
+
+    template_fields = ("location", "workflow_id", "execution_id")
+
+    def __init__(
+        self,
+        *,
+        workflow_id: str,
+        execution_id: str,
+        location: str,
+        project_id: str,
+        success_states: Optional[Set[Execution.State]] = None,
+        failure_states: Optional[Set[Execution.State]] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.success_states = success_states or {Execution.State.SUCCEEDED}
+        self.failure_states = failure_states or {Execution.State.FAILED, Execution.State.CANCELLED}
+        self.workflow_id = workflow_id
+        self.execution_id = execution_id
+        self.location = location
+        self.project_id = project_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+
+    def poke(self, context):
+        hook = WorkflowsHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        self.log.info("Checking state of execution %s for workflow %s", self.execution_id, self.workflow_id)
+        execution: Execution = hook.get_execution(
+            workflow_id=self.workflow_id,
+            execution_id=self.execution_id,
+            location=self.location,
+            project_id=self.project_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+
+        state = execution.state
+        if state in self.failure_states:
+            raise AirflowException(
+                f"Execution {self.execution_id} for workflow {self.execution_id} "
+                f"failed and is in `{state}` state",
```

Comment:
This exception will be raised also when the state is CANCELLED. For that reason I'd change the message to:
```suggestion
                f"finished in `{state}` state",
```